lib/conrad/emitter_queue.rb

```diff
@@ -0,0 +1,52 @@
+module Conrad
+  class EmitterQueue
+    include Singleton
+
+    attr_reader :background
+    attr_accessor :logger
+
+    def initialize
+      Thread.abort_on_exception = true
+      @queue = Queue.new
+      @logger ||= Logger.new(STDOUT)
+    end
+
+    def background=(value)
+      @background = value
+      start_thread if value
```

Comment:
If we're allowing people to flip this back and forth, I think we'll need to do:
```ruby
value ? start_thread : @thread.join
```
to ensure the other thread stops if this gets flipped "off"

Reply:
good call. let me think of how to ent the thread without joining the `loop`